plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1763,144 @@ public void testOptimizeUsingForcedPartitioning()
+        // Drop table to verify register_table call fails when no metadata can be found (table doesn't exist)
+        assertUpdate("DROP TABLE " + tableName);
+
+        assertQueryFails("CALL system.register_table (CURRENT_SCHEMA, '" + tableNameNew + "', '" + tableLocation + "')",
+                ".*Table location (.*) does not exist.*");
+    }
+
+    @Test
+    public void testRegisterTableWithDifferentTableName()
+    {
+        String tableName = "test_register_table_with_different_table_name_old_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 'INDIA', true)", 1);
+
+        String tableLocation = getTableLocation(tableName);
+        // Drop table from metastore and use the table content to register a new table
+        dropTableFromMetastore(tableName);
+
+        String tableNameNew = tableName + "_new";
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableNameNew + "', '" + tableLocation + "')");
+        assertUpdate("INSERT INTO " + tableNameNew + " VALUES(2, 'POLAND', false)", 1);
+
+        assertQuery("SELECT * FROM " + tableNameNew, "VALUES (1, 'INDIA', true), (2, 'POLAND', false)");
+        assertUpdate("DROP TABLE " + tableNameNew);
+    }
+
+    @Test
+    public void testRegisterTableWithNoData()
+    {
+        String tableName = "test_register_table_with_no_data_" + randomNameSuffix();
+
+        assertQuerySucceeds("CREATE TABLE " + tableName + "(a INT, b VARCHAR, c BOOLEAN)");
+
+        String tableLocation = getTableLocation(tableName);
+        String tableNameNew = tableName + "_new";
+
+        assertQuerySucceeds(format("CALL system.register_table('%s', '%s', '%s')", SCHEMA, tableNameNew, tableLocation));
+
+        assertThat(query("SELECT * FROM " + tableName))
+                .matches("SELECT * FROM " + tableNameNew);
+        dropTableFromMetastore(tableNameNew);
+        assertUpdate(format("DROP TABLE %s", tableName));
+    }
+
+    @Test
+    public void testRegisterTableWithTrailingSlashLocation()
+    {
+        String tableName = "test_register_trailing_slash_location_" + randomNameSuffix();
+
+        assertUpdate("CREATE TABLE " + tableName + " (a INT, b VARCHAR, c BOOLEAN)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(1, 'INDIA', true)", 1);
+
+        String tableLocation = getTableLocation(tableName) + "/";
+        // Drop table from metastore and use the same table name to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        assertUpdate("INSERT INTO " + tableName + " VALUES(2, 'POLAND', false)", 1);
+
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true), (2, 'POLAND', false)");
+        assertUpdate("DROP TABLE " + tableName);
+    }
```

Comment:
CC: @findinpath | @findepi | @alexjo2144

Reply:
Is this test run against S3?
What if S3 contains `dir/` directory object (and no `dir` object)?

Reply:
> Is this test run against S3?

Yes, using Minio

> What if S3 contains `dir/` directory object (and no `dir` object)?

S3 does not allow creating `dir/` directory object
<img width="822" alt="image" src="https://user-images.githubusercontent.com/11939923/202499243-6c5c95f6-261d-42c3-a018-35cb0563dc09.png">

Reply:
This test is not required now. table location with and without trailing slash will work without stripping it. I have removed the check on table location existence and instead check transaction log (file) existence.